src/lib/Repository/VersionService.php

```diff
@@ -0,0 +1,37 @@
+<?php
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\HybridPlatformUi\Repository;
+
+use eZ\Publish\API\Repository\ContentService;
+
+class VersionService
+{
+    /**
+     * @var ContentService
+     */
+    private $contentService;
+
+    public function __construct(ContentService $contentService)
+    {
+        $this->contentService = $contentService;
+    }
+
+    /**
+     * Deletes a version based on the contentId and versionNo.
+     *
+     * @param int $contentId
+     * @param int $versionNo
+     */
+    public function deleteVersion(int $contentId, int $versionNo)
+    {
+        $versionInfo = $this->contentService->loadVersionInfo(
+            $this->contentService->loadContentInfo($contentId),
+            $versionNo
+        );
+
```

Comment:
The `contentService` will throw these errors directly so I'd have thought we could catch these directly in the controller or are you suggesting throwing our own errors in here?